bin/charms.reactive.sh

```diff
@@ -1,5 +1,26 @@
 #!/bin/bash
 
+
+{
+    shopt -s expand_aliases
+    alias _suppress_xtrace='{ local old_opts="$-"; set $X_OPT; } 2> /dev/null'
+    alias _restore_xtrace='_restore_opt $old_opts x'
+    function _restore_opt() {
+        if [[ "$1" == *$2* ]]; then
```

Comment:
This if statement was unclear to me at first glance. If I understand this correctly you are looking for the first parameter anywhere in the second parameter to this function. A comment telling me what you are trying to do here would be great.